src/endpoint.jl

```diff
@@ -58,24 +60,56 @@ function Base.:(==)(a::Endpoint, b::Endpoint)
     a.endpoint == b.endpoint && a.included == b.included
 end
 
+function Base.:(==)(a::LeftEndpoint, b::LeftEndpoint)
+    a.endpoint == b.endpoint && a.included == b.included
+end
+
+function Base.:(==)(a::RightEndpoint, b::RightEndpoint)
+    a.endpoint == b.endpoint && a.included == b.included
+end
+
```

Comment:
Both of these are covered by falling back to `Base.:(==)(a::Endpoint, b::Endpoint)`

Reply:
Thats what I wanted to assume, but I was getting these weird issues where it seemed like it wasnt. Perhaps I was somewhat misunderstanding what I was doing. Tests definitely seem to run fine without them as they are falling back as you mentioned.